wrapanapi/virtualcenter.py

```diff
@@ -250,9 +250,16 @@ def _get_vm(self, vm_name, force=False):
         :returns: a pyVmomi object.
         """
         if vm_name not in self._vm_cache or force:
-            vm = self._get_obj(vim.VirtualMachine, vm_name)
-            if not vm:
-                raise VMInstanceNotFound(vm_name)
+            try:
+                vm = self._get_obj(vim.VirtualMachine, vm_name)
+            except vmodl.fault.ManagedObjectNotFound:
+                # TODO this is happening on VMs that still exist, try a second time?
+                self.logger.exception('vmodl.fault.ManagedObjectNotFound calling _get_obj for %s'
+                                      % vm_name)
+                vm = self._get_obj(vim.VirtualMachine, vm_name)
+            finally:
+                if not vm:
```

Comment:
This may raise an unbound local variable error because if both _get_obj invocations fail, vm stays undefined.